### PR TITLE
Add Assets Base URL

### PR DESCRIPTION
**Pull request by LuBuss**

### CODE REVIEW

File: config/server.js [hunk 1]
```diff
@@ -18,14 +18,17 @@ module.exports = {
...
 	adminLoginPath: process.env.ADMIN_LOGIN_PATH || '/admin/login',
 
+	// used by API to service assets
+	assetsBaseURL: process.env.ASSETS_BASE_URL || 'http://localhost',
+
 	apiListenPort: 3001,
-	storeListenPort: 3000,
+	storeListenPort: process.env.STORE_PORT || 3000,
 
 	// used by API
```
Comment (ngoyal16) [inline config/server.js:33]: Port and url is contradict

Running port is 3000
But store url is localhost:3003

File: src/api/server/services/products/images.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ class ProductImagesService {
 					if (product && product.images && product.images.length > 0) {
 						let images = product.images.map(image => {
 							image.url = url.resolve(
-								generalSettings.domain,
+								settings.assetsBaseURL,
```
Comment (ngoyal16) [inline src/api/server/services/products/images.js]: Add domain as alternative to it.

File: src/api/server/services/products/products.js [hunk 3]
```diff
@@ -962,20 +969,20 @@ class ProductsService {
...
+	changeProperties(item, domain, assetsBaseURL) {
 		if (item) {
 			if (item.id) {
 				item.id = item.id.toString();
 			}
 
-			item.images = this.getSortedImagesWithUrls(item, domain);
+			item.images = this.getSortedImagesWithUrls(item, assetsBaseURL);
 
 			if (item.category_id) {
```
Comment (ngoyal16) [inline src/api/server/services/products/products.js:987]: If domain is not used in functions. Don't pass it..

**Review by ngoyal16 — CHANGES_REQUESTED** [review]

File: src/api/server/services/products/products.js (hunk 3 above)
Comment (LuBuss) [reply src/api/server/services/products/products.js:987]: Domain is used in the function to the product url

File: config/server.js (hunk 1 above)
Comment (LuBuss) [reply config/server.js:33]: Updated store url back to 3000. was a late night commit

**Review by ngoyal16 — APPROVED** [review]

**Pull request merged**